modules/core/shared/src/main/scala/lucuma/core/model/Target.scala

```diff
@@ -3,16 +3,32 @@
 
 package lucuma.core.model
 
-import cats._
-
-import lucuma.core.math.ProperMotion
+import scala.collection.immutable.SortedMap
 
-import monocle.macros.Lenses
+import cats._
+import cats.implicits._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.string.NonEmptyString
+import lucuma.core.enum.MagnitudeBand
+import lucuma.core.math._
+import monocle.Lens
+import monocle.Optional
+import monocle.Traversal
+import monocle.function.Each.mapEach
+import monocle.function.FilterIndex
+import monocle.macros.GenLens
+import monocle.std.either._
+import monocle.std.option
 
 /** A target of observation. */
-@Lenses final case class Target(name: String, track: Either[EphemerisKey, ProperMotion])
+final case class Target(
+  name:       NonEmptyString,
+  catalogId:  Option[NonEmptyString],
```

Comment:
This field is to keep a link with the catalogue as the `name` can be changed by the user

Reply:
If it is non-sidereal then the horizons id is in the `EphmerisKey`.  I'm not sure it makes sense for a non-sidereal target.  If this is a sidereal target coming from a catalog then the proper motion data will be coming from the catalog so maybe it makes sense to include it in `ProperMotion`?

Also, should we pair the catalog id with the catalog instead of just a `NonEmptyString`?  Is there an enumeration of catalogs that we could use?  For example:

```
final case class CatalogId(catalog: HypotheticalCatalogEnum, id: NonEmptyString)
```

Reply:
Why do we need to know the catalog id for sidereal targets?